### PR TITLE
[SW-2392] Fix typos in the documentation

### PR DESCRIPTION
**Pull request by jakubhava**
Discovered by applying spell-checker to the doc sources

### CODE REVIEW

File: doc/src/site/sphinx/deployment/aws_emr_edge_node.rst [hunk 1]
```diff
@@ -2,7 +2,7 @@ Use Sparkling Water with Amazon EMR from the Edge Node
 ------------------------------------------------------
 
 Sparkling Water can be deployed on a cluster on EMR or directly on EC2 machines. The jobs can be submitted
-from the edge node which has several security and other benefits.
+from the edge node which have several security and other benefits.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/aws_emr_edge_node.rst:5]: why `have`?

File: doc/src/site/sphinx/deployment/backends.rst [hunk 2]
```diff
@@ -164,9 +164,9 @@ To start an H2O cluster and connect to it, run:
         In case we stored the path of the driver H2O jar to environmental variable ``H2O_DRIVER_JAR``, we don't
         have to specify ``setH2ODriverPath`` as Sparkling Water will read the path from the environmental variable.
 
-When specifying the queue, we recommend that this queue has YARN preemption off in order to have stable a H2O cluster.
+When specifying the queue, we recommend that this queue has YARN preemption off in order to have stable an H2O cluster.
 
-In case of Scala, it can also happen that we might need to explicitly set the client's IP or network. To see how this can be configured, please
+In the case of Scala, it can also happen that we might need to explicitly set the client's IP or network. To see how this can be configured, please
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/backends.rst:169]: why `the`?

File: doc/src/site/sphinx/deployment/kubernetes.rst [hunk 3]
```diff
@@ -10,7 +10,7 @@ Before we start, please check the following:
 
 2. Ensure that we have a working Kubernetes Cluster and ``kubectl`` installed
 
-3. Ensure we have ``SPARK_HOME`` set up to home of our Spark distribution of version SUBST_SPARK_VERSION
+3. Ensure we have ``SPARK_HOME`` set up to the home of our Spark distribution of version SUBST_SPARK_VERSION
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/kubernetes.rst:13]: why `the`?

File: doc/src/site/sphinx/devel/rest_api/scala_interpreter_endpoints.rst [hunk 4]
```diff
@@ -4,15 +4,15 @@ Scala Interpreter via REST API
...
 
 Example usage of Scala Interpreters using the REST API
 ------------------------------------------------------
 
-Here, you can find some basic calls to access scala interpreter behind REST API using curl.
+Here, you can find some basic calls to access Scala interpreter behind REST API using curl.
 
-Init interpreter and obtain session ID
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Init interpreter and obtain asession ID
```
Comment (mn-mikke) [inline doc/src/site/sphinx/devel/rest_api/scala_interpreter_endpoints.rst:14]: `asession`  -> `a session`

File: doc/src/site/sphinx/ml/shap_values.rst [hunk 5]
```diff
@@ -7,7 +7,7 @@ get contributions from raw mojo. The following two sections describe how to achi
 Train model pipeline & get contributions
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Obtaining SHAP values is possible only from H2OGBM, H2OXGBoost and H2ODRF pipeline wrappers and for
+Obtaining SHAP values is possible only from H2OGBM, H2OXGBoost, and H2ODRF pipeline wrappers and for
```
Comment (mn-mikke) [inline doc/src/site/sphinx/ml/shap_values.rst:10]: why `,`?

File: doc/src/site/sphinx/ml/shap_values.rst [hunk 6]
```diff
@@ -124,7 +124,7 @@ To get SHAP values(=contributions) from H2OXGBoost model, please do:
 Get Contributions from Raw MOJO
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Obtaining SHAP values is possible only from MOJO's generated for GBM, XGBoost and DRF and for
+Obtaining SHAP values is possible only from MOJO's generated for GBM, XGBoost, and DRF and for
```
Comment (mn-mikke) [inline doc/src/site/sphinx/ml/shap_values.rst:127]: ditto

File: doc/src/site/sphinx/ml/sw_target_encoder.rst [hunk 7]
```diff
@@ -29,19 +29,19 @@ An example of converting a categorical feature to continues with Target Encoder
   Mountain View   1       0.714    
  =============== ======= =========
 
-Target Encoding can help to improve accuracy of machine learning algorithms when columns with high
-cardinality are used as features during training phase.
+Target Encoding can help to improve the accuracy of machine learning algorithms when columns with high
```
Comment (mn-mikke) [inline doc/src/site/sphinx/ml/sw_target_encoder.rst:32]: why `the` ?

File: doc/src/site/sphinx/pysparkling.rst [hunk 8]
```diff
@@ -10,15 +10,15 @@ This section provides an introduction to PySparkling. To understand more about P
...
+H2O core code is written in Java. Inside H2O, a Distributed Key/Value (DKV) store is used to access and reference data, models, objects, etc., across all nodes/machines, has a non-blocking hashmap, and a memory manager. The algorithms are implemented in the map-reduce style and utilize the Java Fork/Join framework.
 
 The data is read in parallel and distributed across the cluster, stored in memory in a columnar format in a compressed way. H2O's data parser has built-in intelligence to guess the schema of the incoming dataset and supports data ingestion from multiple sources in various formats.
 
-H2O's REST API enables to access all the capabilities of H2O from an external program or script, via JSON over HTTP. The REST API is used by H2O's web interface (Flow UI), the R binding (H2O-R) and the Python binding (H2O-Python).
+H2O's REST API enables to access all the capabilities of H2O from an external program or script, via JSON over HTTP. The REST API is used by H2O's web interface (Flow UI), the R binding (H2O-R), and the Python binding (H2O-Python).
```
Comment (mn-mikke) [inline doc/src/site/sphinx/pysparkling.rst:19]: why `,`?

File: doc/src/site/sphinx/pysparkling.rst [hunk 9]
```diff
@@ -10,15 +10,15 @@ This section provides an introduction to PySparkling. To understand more about P
...
 
-The speed, quality, ease of use and model deployment, for the various cutting-edge supervised and unsupervised algorithms like Deep Learning, Tree Ensembles and Generalized Low Rank Models, makes H2O a highly sought after API for big data analytics.
+The speed, quality, ease of use, and model deployment, for the various cutting-edge supervised and unsupervised algorithms like Deep Learning, Tree Ensembles, and Generalized Low-Rank Models, makes H2O a highly sought after API for big data analytics.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/pysparkling.rst:21]: why `,`?

File: doc/src/site/sphinx/tutorials/hive_support.rst [hunk 10]
```diff
@@ -78,7 +78,7 @@ To import data from non-Kerberized Hive, run:
     .. tab-container:: Scala
         :title: Scala
 
-        First, start Sparkling Shell with the  Hive JDBC client jar on the class path
+        First, start Sparkling Shell with the  Hive JDBC client Jar on the class-path
```
Comment (mn-mikke) [inline doc/src/site/sphinx/tutorials/hive_support.rst:81]: why `Jar`?

File: doc/src/site/sphinx/tutorials/secured_flow.rst [hunk 11]
```diff
@@ -34,7 +34,7 @@ In order to use https correctly, the following two options need to be specified:
             import ai.h2o.sparkling._
             val hc = H2OContext.getOrCreate()
 
-        You can also start Sparkling shell without the configuration and specify it using the setters on ``H2OConf`` as:
+        You can also start a Sparkling shell without the configuration and specify it using the setters on ``H2OConf`` as:
```
Comment (mn-mikke) [inline doc/src/site/sphinx/tutorials/secured_flow.rst:37]: why `a`?

**Comment by jakubhava** [comment]
Thanks for the comments! I have applied this semi-automatically :) It's possible the spell-checker did some mistakes

File: doc/src/site/sphinx/deployment/aws_emr_edge_node.rst (hunk 1 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/aws_emr_edge_node.rst]: Thanks, I connected has with jobs, but the subject is the deployment

File: doc/src/site/sphinx/deployment/backends.rst (hunk 2 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/backends.rst:169]: Seems good to me -> see https://englishstudypage.com/grammar/in-case-of-vs-in-the-case-of-in-english/

File: doc/src/site/sphinx/deployment/kubernetes.rst (hunk 3 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/kubernetes.rst:13]: Should be a in this case

File: doc/src/site/sphinx/devel/rest_api/scala_interpreter_endpoints.rst (hunk 4 above)
Comment (jakubhava) [reply doc/src/site/sphinx/devel/rest_api/scala_interpreter_endpoints.rst:14]: thanks!

File: doc/src/site/sphinx/ml/shap_values.rst (hunk 5 above)
Comment (jakubhava) [reply doc/src/site/sphinx/ml/shap_values.rst:10]: > When you have a list that contains more than two elements, use commas to separate them.

From https://www.grammarly.com/blog/comma/

File: doc/src/site/sphinx/ml/shap_values.rst (hunk 6 above)
Comment (jakubhava) [reply doc/src/site/sphinx/ml/shap_values.rst:127]: Same as above

File: doc/src/site/sphinx/ml/sw_target_encoder.rst (hunk 7 above)
Comment (jakubhava) [reply doc/src/site/sphinx/ml/sw_target_encoder.rst:32]: Should not be here, good catch

File: doc/src/site/sphinx/pysparkling.rst (hunk 8 above)
Comment (jakubhava) [reply doc/src/site/sphinx/pysparkling.rst:19]: Same reason as above ^

File: doc/src/site/sphinx/pysparkling.rst (hunk 9 above)
Comment (jakubhava) [reply doc/src/site/sphinx/pysparkling.rst:21]: dtto

File: doc/src/site/sphinx/tutorials/hive_support.rst (hunk 10 above)
Comment (jakubhava) [reply doc/src/site/sphinx/tutorials/hive_support.rst:81]: Jar is incorrect but should be JAR, thanks

File: doc/src/site/sphinx/tutorials/secured_flow.rst (hunk 11 above)
Comment (jakubhava) [reply doc/src/site/sphinx/tutorials/secured_flow.rst:37]: Good catch, should be `the`

File: doc/src/site/sphinx/deployment/kubernetes.rst (hunk 3 above)
Comment (mn-mikke) [reply doc/src/site/sphinx/deployment/kubernetes.rst]: and why?

File: doc/src/site/sphinx/deployment/backends.rst (hunk 2 above)
Comment (mn-mikke) [reply doc/src/site/sphinx/deployment/backends.rst:169]: ok, thanks!

File: doc/src/site/sphinx/ml/shap_values.rst (hunk 5 above)
Comment (mn-mikke) [reply doc/src/site/sphinx/ml/shap_values.rst:10]: ok, thanks!

File: doc/src/site/sphinx/pysparkling.rst (hunk 8 above)
Comment (mn-mikke) [reply doc/src/site/sphinx/pysparkling.rst:19]: thanks

File: doc/src/site/sphinx/deployment/kubernetes.rst (hunk 3 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/kubernetes.rst]: My understanding is that the user can have multiple spark home directories. By using `the` we refer to specific home directory.

**Pull request merged**